### DISCUSSION TITLE
Downloading TommyPROM sketch

### DISCUSSION THREAD

**Discussion by prlaba** [TomNisbet/TommyPROM#49] [General]
Since the Arduino IDE always runs its current sketch when powered on, is it possible that the EEPROM being programmed or the shift registers could be accidentally damaged when downloading the TommyPROM sketch into a Nano with another sketch already installed (and running)?

My Nano came with a “blinker” sketch pre-installed (toggles the onboard D13 LED on and off at 1-sec Intervals), so that's unlikely to cause any problems.  But a different sketch in the Nano might have an I/O data pin defined as OUTPUT connected to the EEPROM’s input pin.  Ditto for the shift register ICs.

Should I be concerned about this possibility?

**Comment by TomNisbet** [comment]
There wouldn't be any issue with the shift registers. On the EEPROM side, you'd probably only get in trouble if the Arduino was defining both CE and WE as an outputs and pulling them both low while also asserting signals on the data lines. The default blink sketch won't do this, but it's possible if you had a different sketch loaded previously.  I build my TommyPROMs with pulp resistors on the three EEPROM signals so there are are disabled if the Arduino leaves them floating.  It's also good practice to burn an initial version of the code into the Arduino before installing it.
